code/game/objects/structures/lavaland/ore_vent.dm

```diff
@@ -0,0 +1,502 @@
+	)
+	///What items can be used to scan a vent?
+	var/static/list/scanning_equipment = list(
+		/obj/item/t_scanner/adv_mining_scanner,
+		/obj/item/mining_scanner,
+	)
+
+	/// What base icon_state do we use for this vent's boulders?
+	var/boulder_icon_state = "boulder"
+	/// Percent chance that this vent will produce an artifact boulder.
+	var/artifact_chance = 0
+	/// We use a cooldown to prevent the wave defense from being started multiple times.
+	COOLDOWN_DECLARE(wave_cooldown)
+
+
+/obj/structure/ore_vent/Initialize(mapload)
+	generate_description()
+	register_context()
+	SSore_generation.possible_vents += src
+	boulder_icon_state = pick(list(
+		"boulder",
+		"rock",
+		"stone",
+	))
+	if(tapped)
+		SSore_generation.processed_vents += src
+		icon_state = icon_state_tapped
+		update_appearance(UPDATE_ICON_STATE)
+		add_overlay(mutable_appearance('icons/obj/mining_zones/terrain.dmi', "well", ABOVE_MOB_LAYER, src, GAME_PLANE_UPPER))
+	return ..()
+
+/obj/structure/ore_vent/Destroy()
+	SSore_generation.possible_vents -= src
+	node = null
+	if(tapped)
+		SSore_generation.processed_vents -= src
+	return ..()
+
+/obj/structure/ore_vent/attackby(obj/item/attacking_item, mob/user, params)
+	. = ..()
+	if(.)
+		return
+	if(!is_type_in_list(attacking_item, scanning_equipment))
+		return
+	if(tapped)
+		balloon_alert_to_viewers("vent tapped!")
+		return TRUE
+	scan_and_confirm(user)
+	return TRUE
+
+/obj/structure/ore_vent/attack_hand(mob/living/user, list/modifiers)
+	. = ..()
+	if(.)
+		return
+	if(!isgolem(user))
+		return
+	if(!discovered)
+		to_chat(user, span_notice("You can't quite find the weakpoint of [src]... Perhaps it needs to be scanned first?"))
+		return TRUE
+	to_chat(user, span_notice("You start striking [src] with your golem's fist, attempting to dredge up a boulder..."))
+	for(var/i in 1 to 3)
+		if(do_after(user, boulder_size * 1 SECONDS, src))
+			user.apply_damage(20, STAMINA)
+			playsound(src, 'sound/weapons/genhit.ogg', 50, TRUE)
+	produce_boulder()
+	visible_message(span_notice("You've successfully produced a boulder! Boy are your arms tired."))
+	return TRUE
+
+/obj/structure/ore_vent/is_buckle_possible(mob/living/target, force, check_loc)
+	. = ..()
+	if(tapped)
+		return FALSE
+	if(istype(target, /mob/living/basic/node_drone))
+		return TRUE
+
+/obj/structure/ore_vent/examine(mob/user)
+	. = ..()
+	if(discovered)
+		switch(boulder_size)
+			if(BOULDER_SIZE_SMALL)
+				. += span_notice("This vent produces [span_bold("small")] boulders containing [ore_string]")
+			if(BOULDER_SIZE_MEDIUM)
+				. += span_notice("This vent produces [span_bold("medium")] boulders containing [ore_string]")
+			if(BOULDER_SIZE_LARGE)
+				. += span_notice("This vent produces [span_bold("large")] boulders containing [ore_string]")
+	else
+		. += span_notice("This vent can be scanned with a [span_bold("Mining Scanner")].")
+
+/obj/structure/ore_vent/add_context(atom/source, list/context, obj/item/held_item, mob/living/user)
```

Comment:
For this to work you have to call `register_context()` inside `Initialize()`

Reply:
Right, I am on line 77.